File: our-life/src/app/pojo/recipe.ts
import { Ingredient } from "./ingredient";
import { Nutrition } from "./nutrition";
import { Step } from "./step";
import { SugarReading } from "./sugar-reading";

export interface Recipe {
    id:string,
    name:string;
    description?:string;
    url:string;
    ingredients:Ingredient[];
    preparation:Step[];
    cooking:Step[];
    sugarReadings: SugarReading[];
    nutrition:Nutrition;
}

File: identity-server/src/main/java/com/avasthi/varahamihir/identityserver/entities/Recipe.java
package com.avasthi.varahamihir.identityserver.entities;

import java.util.List;
import java.util.UUID;

public record Recipe(UUID id,
                     String name,
                     String description,
                     String url,
                     List<IngredientEntity> ingredients,
                     List<Step> preparation,
                     List<Step> cooking,
                     List<SugarReading> sugarReadings,
                     Nutrition nutrition) {
}
